connect-inject/patch.go

```diff
@@ -1,131 +0,0 @@
-package connectinject
```

Comment:
everything here was unused, except for `escapeJSONPointer`, which was only used in tests. I've moved it to the test that was using it instead.